PXC-3767: IPv6 support in PXC (wsrep_sst_xtrabackup-v2)

nitendra-bhosle opened the pull request:
https://jira.percona.com/browse/PXC-3767

Analysis
Joiner nodes are failing , becuase they are using
socat utility with IPv6 and currently wsrep_sst_xtrabackup-v2 script
doesn't support socat command with IPv6 option.

Fix
Added IPv6 support for socat command inside wsrep_sst_xtrabackup-v2 script
for Joiner nodes.

nitendra-bhosle commented:
Jenkins Run :  https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/345/

dutow reviewed (commented):
The change itself looks good, but shouldn't we fix this also for 5.7?

nitendra-bhosle commented:
> The change itself looks good, but shouldn't we fix this also for 5.7?

I have asked question on JIRA . If reporter asks for both 5.7 and 8.0 , I will create review request on 5.7 as well

Commit:
PXC-3767: IPv6 support in PXC (wsrep_sst_xtrabackup-v2)

https://jira.percona.com/browse/PXC-3767

Analysis
Joiner nodes are failing , becuase they are using
socat utility with IPv6 and currently wsrep_sst_xtrabackup-v2 script
doesn't support socat command with IPv6 option.

Fix
Added IPv6 support for socat command inside wsrep_sst_xtrabackup-v2 script
for Joiner nodes.

kamil-holubicki reviewed (approved):
lgtm
but, shouldn't we use GCA process for this and https://github.com/percona/percona-xtradb-cluster/pull/1527?

nitendra-bhosle commented:
> lgtm but, shouldn't we use GCA process for this and #1527?

I have followed GCA process for https://github.com/percona/percona-xtradb-cluster/pull/1527 
Once this will be approved and merged ,  I will update this PR with commit from GCA

nitendra-bhosle commented:
New Jenkins : https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/391/

kamil-holubicki reviewed (commented):
Still does not follow GCA

Commit:
PXC-3767: IPv6 support in PXC (wsrep_sst_xtrabackup-v2)

https://jira.percona.com/browse/PXC-3767

Merge remote-tracking branch 'origin/5.7-PXC-3767' into 8.0-PXC-3767

kamil-holubicki reviewed (approved):
lgtm
